core/trino-main/src/main/java/io/trino/execution/scheduler/PartitioningHandlePartitionCount.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.scheduler;
+
+import io.trino.sql.planner.PartitioningHandle;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+final class PartitioningHandlePartitionCount
```

Comment:
I think we can have this class directly inside `PipelineQueryScheduler`

Reply:
Maybe use record?

Reply:
Also rename this to `PartitioningKey`

Reply:
doneZ